mitodl/mit-learn
# Query membership endpoints for resources belonging to user lists and learning paths

### What are the relevant tickets?





Closes https://github.com/mitodl/hq/issues/6032

### Description (What does it do?)


Provides hooks for components to determine a resource's User List and Learning Path memberships. This replaces relationships being specified on resources in their `user_list_parents` and `learning_path_parents` arrays, so we have been able to remove all the query cache invalidation logic that determined which resources were stale when change were made to the parent lists. We now just invalidate the respective membershipList caches (on userLists and learningPaths).

### How can this be tested?


Resource card and learning drawer action buttons should highlight according to User List and Learning Path memberships and should be responsive to changes made in the Add to List dialogs.

Changes made in the Add to List dialogs in one screen should be reflected in other screens while navigating without page refresh (Homepage, Search, User List (/dashboard/my-lists/1) and Learning Path details (/learningpaths/1) pages.

If the only User List or Learning Path that a Learning Resource is a member of is deleted, the action button should no longer be highlighted.


### Additional Context



The membership lists previously lived on `user_list_parents` and `learning_path_parents` on learning resources, however these are specific to a user so cannot be cached on public CDN. Membership lists endpoints have been added to the API with https://github.com/mitodl/mit-learn/pull/1808 so that they can be queried for independently of the learning resources.

These do still exist on the API and can be cleaned out, but for now are overriden to `[]` in the query hooks to ensure no remaining stray dependency.
- /api/v1/learning_resources/{id}/
- /api/v1/learning_resources/
- /api/v1/featured/
- /api/v1/learning_resources_search/
- /api/v1/userlists/{userlist_id}/items/
- /api/v1/learningpaths/{learning_resource_id}/items/


Also fixes (by bypass) an issue in Prod where resources are not bookmarked in the User List details screen as `user_list_parents` on /api/v1/userlists/{userlist_id}/items/ is always empty.
